File: app/src-tauri/src/models/http.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HttpResponse {
    pub url: String,
    pub status_code: u16,
    pub final_url: String,
    pub redirects: Vec<HttpRedirect>,
    pub headers: HashMap<String, String>,
    pub response_time: f64,
    pub raw_output: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HttpRedirect {
    pub from_url: String,
    pub to_url: String,
    pub status_code: u16,
}

File: src/models/http.ts
export interface HttpRedirect {
  from_url: string;
  to_url: string;
  status_code: number;
  response_time: number;
}

export interface HttpResponse {
  url: string;
  status_code: number;
  final_url: string;
  redirects: HttpRedirect[];
  headers: Record<string, string>;
  response_time: number;
  raw_output?: string;
}
